apps/hash-graph/lib/graph/src/store/fetcher.rs

```diff
@@ -196,106 +203,113 @@ where
 
     async fn fetch_external_ontology_types(
         &self,
-        ontology_type_references: impl IntoIterator<
-            Item = (OntologyTypeReference<'_>, UpdatedById),
-            IntoIter: Send,
-        > + Send,
-    ) -> Result<
-        (
-            Vec<(DataType, OntologyElementMetadata)>,
-            Vec<(PropertyType, OntologyElementMetadata)>,
-            Vec<(EntityType, OntologyElementMetadata)>,
-        ),
-        StoreError,
-    > {
-        let fetcher_client = self.fetcher_client().await?;
-
-        let mut data_types = Vec::new();
-        let mut property_types = Vec::new();
-        let mut entity_types = Vec::new();
-
-        for (reference, fetched_by) in ontology_type_references {
-            let provenance_metadata = ProvenanceMetadata::new(fetched_by);
-            let fetched_ontology_types = fetcher_client
-                .fetch_ontology_type_exhaustive(context::current(), reference.url().clone())
+        ontology_type_references: Vec<VersionedUrl>,
+        actor_id: UpdatedById,
+    ) -> Result<FetchedOntologyTypes, StoreError> {
+        let provenance_metadata = ProvenanceMetadata::new(actor_id);
+
+        let mut queue = ontology_type_references;
+        let mut seen = queue.iter().cloned().collect::<HashSet<_>>();
+
+        let mut fetched_ontology_types = FetchedOntologyTypes::default();
```

Comment:
Previously, we used `stream::unfold`, but I chose the manual approach here as this way I can directly add them to the three different sets for ontology types, `unfold` only supports a single collection. We could implement the additional traits for `FetchedOntologyTypes`, but as it's only used here it's not worth the effort.